### ISSUE TITLE
App hangs after sleep mode

### ISSUE THREAD

**Issue by rince1984** [neutralinojs/neutralinojs#1021]
**Describe the bug**
The neutralino app runs under windows and the pc changes to sleep mode (e.g. close notebook). On Returning from sleep mode to windows (e.g. open notebook) the app hangs. Its not possible to close the app can not be closed.

Windows event manager say:
```
The program neutralino-win_x64.exe version 0.0.0 has stopped interacting with Windows and has been closed. Check the problem history in the Security and Maintenance control panel to look for more information about the problem.
 Process ID: 29c0
 Start time: 01d8f0f637f54100
 Exit time: 4294967295
 Application path: C:\SCRIPTS\APPS\NEUTRALINOJS\myApp\bin\neutralino-win_x64.exe
 Bericht-ID: 8051b203-3380-465f-851d-6d047577032d
 Full name of the corrupted package: 
 Relative application ID of the corrupted package: 
 Crash type: Top level window is idle
```

**To Reproduce**
1. Start app (e.g. `neu run`)
2. Close notebook or go to sleep mode
3. Open notebook
4. Try closing

**Specifications**  
 - OS: Windows 10 x64
 - Neutralinojs version: v4.8.0
 - Neutralinojs client library version: v3.7.0
 - Neutralinojs CLI version: v9.2.0

**Comment by Bonnev** [comment]
Not only closing, the whole websocket connection between the webapp and the server terminates. This is a big problem and basically means we cannot use Neutralino if our computer will go to sleep.

**Comment by rince1984** [comment]
> Not only closing, the whole websocket connection between the webapp and the server terminates, so you cannot write to files, you cannot open processes, you cannot communicate with the OS at all. This is a big problem and basically means we cannot use Neutralino if our computer will go to sleep.

I hope that issue will be addressed. For my current, small apps it is not critical. However, I am planning to develop a larger, productive app for customers and there it would be fatal. I love Neutralinojs, but I'm currently thinking about going back to NWJS for the new project.

This issue makes neu apps instable.

**Comment by shalithasuranga** [comment]
Hey @rince1984, thanks for reporting this issue, I think we can try to solve this issue and apply a fix with the next release as a priority since it breaks working apps when app users do a action (closing the laptop lid) they typically do

**Comment by globules-io** [comment]
Any update on this? Thanks

**Comment by rince1984** [comment]
For all those who also have this problem. I have integrated the following workaround in my apps. It checks if the server goes offline (Neutralinojs event) and if it does, I do a refresh (```window.location.reload```). Of course everything that was entered before the standby (energy saving mode) is lost. Not nice, but acceptable as a workaround for my small apps.

```
Neutralino.events.on("serverOffline",function(){
	var date = new Date();
	var timestamp = date.toLocaleString('en-US');
	console.log('server offline ' + timestamp );
	window.location.reload();
});
```

I have tried it several times on my machine (Win 10 Pro, Desktop PC) and it worked fine, but I do not know if it works on Win11 etc.

**Comment by globules-io** [comment]
@rince1984 thanks for sharing, we will use that as a temp work around

**Comment by ivocass** [comment]
> Was a fix ever implemented?

No. When you come back from sleep, the Neutralino app is disconnected from the server.
So only the app's front-end works.
Even actions from the tray menu like Restart, Quit, Get version, all fail silently.
What's worse, **you can't write the app's state to disk**, as localStorage doesnt' persist, and Neutralino.storage isn't accessible.
It's a shame, since I developed an app thinking Neutralino was mature.

**Comment by ivocass** [comment]
> Was a fix ever implemented?

No. When you come back from sleep, the Neutralino app is disconnected from the server.
So only the app's front-end works.
Even actions from the tray menu like Restart, Quit, Get version, all fail silently.

The only solution right now is to precisely save the app's state to localStorage, reload the window, then load the state.
And make sure you delay the window reloading with a short timeout. Otherwise your keyboard and mouse listeners will throw errors.

**Comment by shalithasuranga** [comment]
Closing as a duplicate of #970. Thanks everyone :)
